fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AstToSQLBuilderTest.java

```diff
@@ -63,33 +63,127 @@ public void testInsertFromFiles() {
         StatementBase stmt = SqlParser.parseSingleStatement(sql, SqlModeHelper.MODE_DEFAULT);
         Assertions.assertEquals(
                 "INSERT INTO `t0` (`v1`,`v2`) " +
-                        "SELECT *\nFROM FILES(\"aws.s3.access_key\" = \"***\", \"aws.s3.region\" = \"us-west-1\", " +
+                        "SELECT \n  *\nFROM FILES(\"aws.s3.access_key\" = \"***\", \"aws.s3.region\" = \"us-west-1\", " +
                         "\"aws.s3.secret_key\" = \"***\", \"format\" = \"parquet\", \"path\" = \"s3://xxx/zzz\")",
                 AstToSQLBuilder.toSQL(stmt));
     }
     @Test
     public void testSelectStarExcludeToSQL() throws Exception {
         String sql = "SELECT * EXCLUDE (name, email) FROM test_exclude;";
         StatementBase stmt = SqlParser.parseSingleStatement(sql, SqlModeHelper.MODE_DEFAULT);
-        Assertions.assertEquals("SELECT * EXCLUDE ( \"name\",\"email\" ) \nFROM `test_exclude`",
+        Assertions.assertEquals("SELECT \n  * EXCLUDE ( \"name\",\"email\" ) \nFROM `test_exclude`",
                 AstToSQLBuilder.toSQL(stmt));
         
         sql = "SELECT test_exclude.* EXCLUDE (name) FROM test_exclude";
         stmt = SqlParser.parseSingleStatement(sql, SqlModeHelper.MODE_DEFAULT);
-        Assertions.assertEquals("SELECT test_exclude.* EXCLUDE ( \"name\" ) \nFROM `test_exclude`",
+        Assertions.assertEquals("SELECT \n  test_exclude.* EXCLUDE ( \"name\" ) \nFROM `test_exclude`",
                 AstToSQLBuilder.toSQL(stmt));
     }
 
     @Test
     public void testFunctionTable() {
         String sql = "SELECT * from tarray, unnest(v3) as t(x)";
         StatementBase stmt = SqlParser.parseSingleStatement(sql, SqlModeHelper.MODE_DEFAULT);
-        Assertions.assertEquals("SELECT *\nFROM `tarray` , unnest(`v3`) t(`x`) ",
+        Assertions.assertEquals("SELECT \n  *\nFROM `tarray` , unnest(`v3`) t(`x`) ",
                 AstToSQLBuilder.toSQL(stmt));
 
         sql = "SELECT * from t0, generate_series(v1, v2, 1) as t(x)";
         stmt = SqlParser.parseSingleStatement(sql, SqlModeHelper.MODE_DEFAULT);
-        Assertions.assertEquals("SELECT *\nFROM `t0` , generate_series(`v1`,`v2`,1) t(`x`) ",
+        Assertions.assertEquals("SELECT \n  *\nFROM `t0` , generate_series(`v1`,`v2`,1) t(`x`) ",
                 AstToSQLBuilder.toSQL(stmt));
     }
+
+
+    @Test
+    public void testCaseWhenFormatting() {
+        String sql = "SELECT CASE WHEN v1 < 10 THEN 'low' WHEN v1 >= 10 AND v1 < 20 THEN 'medium' ELSE 'high' END FROM t0";
+        StatementBase stmt = SqlParser.parseSingleStatement(sql, SqlModeHelper.MODE_DEFAULT);
+        String expected = "SELECT \n" +
```

Comment:
other syntax uses the `XXX\n` rather than `XXX \n`, so for consistency we should remove the ` ` here